shared/cluster.go

```diff
@@ -633,3 +633,64 @@ func FetchToken(ip string) (string, error) {
 
 	return token, nil
 }
+
+func CreateSecret(secret, namespace string) error {
+	kubectl := fmt.Sprintf("kubectl --kubeconfig %s", KubeConfigFile)
+	if namespace == "" {
+		namespace = "default"
+	}
+	cmd := fmt.Sprintf("%s create secret generic %s -n %s --from-literal=mykey=mydata",
+		kubectl, secret, namespace)
+	createStdOut, err := RunCommandHost(cmd)
+	if err != nil {
+		return ReturnLogError("failed to create secret: \n%w", err)
+	}
+	if strings.Contains(createStdOut, "failed to create secret") {
+		return ReturnLogError("failed to create secret: \n%w", err)
+	}
+	LogLevel("DEBUG", "Create Secret Output: %s", createStdOut)
+	return nil
+}
+
+func checkPodStatus(print bool) bool {
```

Comment:
https://github.com/rancher/distros-test-framework/blob/e189754e84556cbd9d0c9b3b695967d07c665e01/pkg/assert/pod.go#L61
https://github.com/rancher/distros-test-framework/blob/e189754e84556cbd9d0c9b3b695967d07c665e01/pkg/assert/pod.go#L121

Reply:
Both of them use Expects - which means, they will fail. These waits are meant for NOT waiting too long, and NOT erroring/failing. It is meant to be forgiving. I cannot use either of those for this reason.

Reply:
ok but the usage of this func which is on the `WaitForPodsRunning` on line 696 
```
if !podsRunning {
		return ReturnLogError("All pods were not up at the end of wait period %d", int(defaultTime)*times)
	}
```

you are returning error if they are not up in that period of time, which is exactly what those functions do,  right ?

if your concern its the duration , thats a design discussion now to improve that, not duplicating code .

if am still missing something please let me know.

Reply:
idk why but my comments was not uploaded here 

so , the user of this is `WaitForPodsRunning` its returning error here if the pods are not running, which is exact the same thing on the exist functions already do.  if your concern its the time duration, this should be a design enhancement but not creating extra stuff duplicating and doing exact the same only changing the duration. 
```
	}
	if !podsRunning {
		return ReturnLogError("All pods were not up at the end of wait period %d", int(defaultTime)*times)
	}
```

Reply:
i think i am not understanding well the requirement sorry 
 can u explain me? why then u need all of this, if you are actually doing nothing with the result ( success or not ) ?

Reply:
This came from my ask -- the reasoning was so that we could have a "wait up to X seconds for this condition" as previously this was just "sleep X" where X was an arbitrary number that had worked in local tests but might not work always.

PodStatusRunning assertion does wait up to 30 seconds, so it's possible that we can still use that but maybe increase the timeout on that function to match what is needed here in the maximum case. I'm fine either way and will let @aganesh-suse decide what's best there.

Reply:
We had a hard-coded wait time before. Max suggested we avoid this. 
We need time buffers between node restarts. And post node restart - we have to wait before we run the secrets-encryption test. Otherwise the tests will fail. How do we quantify this time? How long a wait is a good time? 
We could query for the pods to come up. - But this is not enough. Since the secrets-encryption processes after the pods come up. So then we query for the hash match as well - before we start the test. This ends up being a wait for pods running + wait for hash match. 

We are waiting till the the pods come up. If they come up quickly, `we save some wait time` and can get on with the test faster.  Instead of a blind wait for 30 seconds say, we now `monitor & log` - which reduces wait time so far to like 15 to 20 seconds in different test runs. 
Also, this is a way to simply `know the pod status before we started the secret-encryption test`. 
 If they dont come up by the time, that is fine too. since we will do restarts anyway for other servers. 
Plus we have a wait time for the hash match later - which will be enough time for the pods to have come up. 

Also, at the end we have the Test for Pod Status - which will cover the error for pods not up after secret encryption.

Reply:
ok , if we are using this a shared stuff , we need to contemplate other conditionals as well like the ones from here 
https://github.com/rancher/distros-test-framework/blob/e189754e84556cbd9d0c9b3b695967d07c665e01/pkg/testcase/pod.go#L34

Reply:
The etcd node pods wont come up until the control node plane joins the cluster. If we assert fail for pods running here, it would have been premature - and we would have a failed test for a non-bug.

Reply:
so if this  is too specific for your use case, this 2 funcs should live on the test case file, not here.

Reply:
I feel like it's fine in this cluster.go file in this case -- since this is a shared function that is at cluster level operations. It's not really a testcase since there aren't any `Expect` statements

Reply:
not a test case but a specific auxiliar to it ...so should live in the same file

if we have at least one other context,scneario,test that could use this func, then its ok to be here. but as far i am seeing, difficult to other place be using it, except the encryp

Reply:
so here , on shared pkg we should have functions that can be used in elsewhere, this function in conjunction with `WaitForPodsRunning` can we use in other places ? 
because as i can see its a very specific use case for the encryption test right ? 

@rancher-max  @aganesh-suse